app/packages/core/src/plugins/SchemaIO/components/FileExplorerView/state.ts

```diff
@@ -208,6 +222,20 @@ export function useFileExplorer(fsInfo, chooseMode, onChoose) {
     setCurrentDirectory(null);
   };
 
+  const handleCustomPathChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
Moving some of this logic into the path util would make it testable and could give us confidence that the various path styles supports this.